cloudify_cli/tests/commands/test_deployments.py

```diff
@@ -950,3 +952,305 @@ def test_deployment_modifications_get(self):
         assert 'Removed node instances:' not in output_lines
         added_title_idx = output_lines.index('Added node instances:')
         assert 'node1_olbbe0 (node1)' in output_lines[added_title_idx + 1]
+
+
+class DeploymentScheduleTest(CliCommandTest):
+    def setUp(self):
+        super(DeploymentScheduleTest, self).setUp()
+        self.use_manager()
+
+    def test_deployment_schedule_create(self):
+        self.client.execution_schedules.create = MagicMock(
+            return_value=execution_schedules.ExecutionSchedule({}))
+        self.invoke('cfy deployments schedule create backup -d dep1 '
+                    '-s "12:00" -u "+1w +1d" -r 2d --tz EST')
+
+        expected_since = datetime.datetime.utcnow().replace(
+            hour=17, minute=0, second=0, microsecond=0)
+        expected_until = \
+            (datetime.datetime.utcnow().replace(second=0, microsecond=0) +
+             datetime.timedelta(days=8))
+
+        call_args = list(self.client.execution_schedules.create.call_args)
+        assert call_args[0][0] == 'dep1_backup'
+        assert call_args[1]['since'] == expected_since
+        assert call_args[1]['until'] == expected_until
+        assert call_args[1]['frequency'] == '2d'
+
+    def test_deployment_schedule_create_with_schedule_name(self):
+        self.client.execution_schedules.create = MagicMock(
+            return_value=execution_schedules.ExecutionSchedule({}))
+        self.invoke('cfy deployments schedule create backup -d dep1 -n '
+                    'back_me_up -s "1905-6-13 12:00" --tz GMT')
+
+        expected_since = \
+            datetime.datetime.strptime('1905-6-13 12:00', '%Y-%m-%d %H:%M')
+
+        call_args = list(self.client.execution_schedules.create.call_args)
+        assert call_args[0][0] == 'back_me_up'
+        assert call_args[1]['since'] == expected_since
+        assert not call_args[1]['frequency']
+        assert not call_args[1]['until']
+
+    def test_deployment_schedule_create_default_timezone_local(self):
+        self.client.execution_schedules.create = MagicMock(
+            return_value=execution_schedules.ExecutionSchedule({}))
+        self.invoke('cfy deployments schedule create install -d dep2 '
+                    '-s "1905-6-13 9:00"')
+
+        expected_since = \
+            (datetime.datetime.strptime('1905-6-13 09:00', '%Y-%m-%d %H:%M') -
+             (datetime.datetime.now() - datetime.datetime.utcnow()))
+
+        call_args = list(self.client.execution_schedules.create.call_args)
+        assert call_args[0][0] == 'dep2_install'
+        assert (call_args[1]['since'] - expected_since).total_seconds() < 0.1
```

Comment:
unfortunately, without resorting to more external libraries, finding out the machine's local timezone is very tricky. 
I didn't want to use the exact same way in the test that I do in the function, so that's what I came up with

Reply:
It doesn't make much sense to me. So this test basically tests two different implementations for getting the timezone of the machine? How can we be sure that one method is better than the other?